src/net_processing.cpp

```diff
@@ -2565,14 +2565,22 @@ bool PeerManagerImpl::TryLowWorkHeadersSync(Peer& peer, CNode& pfrom, const CBlo
 
             // Now a HeadersSyncState object for tracking this synchronization is created,
             // process the headers using it as normal.
-            return IsContinuationOfLowWorkHeadersSync(peer, pfrom, headers);
+            if (!IsContinuationOfLowWorkHeadersSync(peer, pfrom, headers)) {
+                // Something went wrong, reset the headers sync.
+                peer.m_headers_sync.reset(nullptr);
+                LOCK(m_headers_presync_mutex);
+                m_headers_presync_stats.erase(peer.m_id);
+            }
```

Comment:
It looks like we don't need this block of code that clears out `peer.m_headers_sync` after all, because this is handled in `IsContinuationOfLowWorkHeadersSync`.  So this could just be:
```
(void) IsContinuationOfLowWorkHeadersSync(peer, pfrom, headers);
```
with a comment explaining that failures are handled inside that function, and that we know there is nothing more to be done even if the function succeeds.

Reply:
Done in the follow up PR https://github.com/bitcoin/bitcoin/pull/26387.